src/rt/lifetime.d

```diff
@@ -2036,6 +2039,30 @@ extern (C) void[] _d_arrayappendcd(ref byte[] x, dchar c)
     return _d_arrayappendT(typeid(shared char[]), x, appendthis);
 }
 
+unittest
+{
+    import core.exception : UnicodeException;
+
+    static void assertThrown(T : Throwable = Exception, E)(lazy E expr, string msg)
+    {
+        try
+            expr;
+        catch (T e) {
+            assert(e.msg == msg);
+            return;
+        }
+    }
```

Comment:
OK, please add a comment summarizing why it was necessary to add this template instead of just using the inline code.